src/main/java/org/myrobotlab/net/Pinger.java

```diff
@@ -0,0 +1,65 @@
+package org.myrobotlab.net;
+
+import java.net.InetAddress;
+import java.util.Map;
+
+import org.myrobotlab.framework.interfaces.Invoker;
+import org.myrobotlab.logging.LoggerFactory;
+import org.slf4j.Logger;
+
+public class Pinger implements Runnable {
+
+  public final static Logger log = LoggerFactory.getLogger(Pinger.class);
+
+  protected String ip;
+  protected int timeout;
+  protected Invoker invoker;
+  protected Map<String, Host> hosts;
+
+  public Pinger(Invoker invoker, Map<String, Host> hosts, String ip, int timeout) {
+    this.ip = ip;
+    this.timeout = timeout;
+    this.invoker = invoker;
+    this.hosts = hosts;
+  }
+
+  @Override
+  public void run() {
+    try {
+      if (InetAddress.getByName(ip).isReachable(timeout)) {
+        Host h = null;
+        if (hosts.containsKey(ip)) {
+          h = hosts.get(ip);
+        } else {
+          h = new Host();
+          h.ip = ip;
+          h.name = "new-host";
```

Comment:
we probably shouldn't hard code the host name to "new-host" here ?  if we assign it later when we know it, we could set it to a default value of unknown or something?

Reply:
the data from the Pinger comes from ICMP - you don't get host name .. "new-host" is currently the default